pkgs/applications/networking/instant-messengers/riot/riot-web.nix

```diff
@@ -4,11 +4,13 @@
 # Versions of `riot-web` and `riot-desktop` should be kept in sync.
 
 let
-  noPhoningHome = {
+  privacyOverrides = writeText "riot-config-privacy.json" (builtins.toJSON {
     disable_guests = true; # disable automatic guest account registration at matrix.org
     piwik = false; # disable analytics
-  };
-  configOverrides = writeText "riot-config-overrides.json" (builtins.toJSON (noPhoningHome // conf));
+  });
+  userOverrides = writeText "riot-config-user.json" (
+    with builtins; if isAttrs conf then toJSON conf else conf
```

Comment:
I'm not a fan of options that have no proper type.
I think it would be better to add a `configFile` option for this: it would still be a breaking change but it would only happen in the next relase, with

Reply:
We could only backport this PR and leave `attrset`-only for the future.